ddtrace/api.py

```diff
@@ -55,8 +82,8 @@ def send_traces(self, traces):
         response = self._put(self._traces, data, len(traces))
 
         # the API endpoint is not available so we should downgrade the connection and re-try the call
-        if response.status in [404, 415] and self._compatibility_mode is False:
-            log.debug('calling the endpoint "%s" but received %s; downgrading the API', self._traces, response.status)
+        if response.status in [404, 415] and self._fallback:
```

Comment:
There @palazzem I'd like to have your opinion on this. I thought it was OK to cascade down from v0.4 to v0.3 to v0.2 but I'm open to other possibilities. The advantage of doing this is that here I just care about the endpoint, but the question of knowing wether we have a JSON as an answer or a plain `OK\n` string is done later. This enables low coupling between this chunk of code and the one handling the JSON.